File: BackEnd/src/main/java/cat/copernic/ymelero/EntreBicis/entity/Recompensa.java
package cat.copernic.ymelero.entrebicis.entity;

import java.time.LocalDate;

import org.springframework.format.annotation.DateTimeFormat;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Lob;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "recompensa")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Recompensa {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String descripcio;

    private String observacions;

    private Double punts;

    @Enumerated(EnumType.STRING)
    private EstatRecompensa estat;

    @Column(name = "nom_comerç")
    private String nomComerç;

    @Column(name = "adreça_comerç")
    private String adreçaComerç;

    @Lob
    @Column(columnDefinition = "MEDIUMBLOB")
    private byte[] foto;

    @ManyToOne
    private Usuari usuari;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @Column(name = "data_reserva")
    private LocalDate dataReserva;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @Column(name = "data_recollida")
    private LocalDate dataRecollida;
}

File: FrontEnd/app/src/main/java/cat/copernic/ymelero/entrebicis/core/model/Recompensa.kt
package cat.copernic.ymelero.entrebicis.core.model

data class Recompensa (
    val id: Long,
    val descripcio: String,
    val observacions: String?,
    val punts: Double,
    val estat: EstatRecompensa,
    val nomComerc: String,
    val adrecaComerc: String,
    val foto: String? = null,
    val usuari: Usuari? = null,
    val dataReserva: String? = null,
    val dataRecollida: String? = null,
    val dataCreacio: String? = null,
    val dataAssignacio: String? = null,
)